hail/python/hailtop/hailctl/batch/submit.py

```diff
@@ -0,0 +1,73 @@
+import asyncio
+import orjson
+import os
+
+import hailtop.batch as hb
+import hailtop.batch_client.client as bc
+from hailtop import pip_version
+from hailtop.aiotools.copy import copy_from_dict
+from hailtop.config import get_remote_tmpdir, get_user_config_path, get_deploy_config
+from hailtop.utils import unpack_comma_delimited_inputs
+
+HAIL_GENETICS_HAIL_IMAGE = os.environ.get('HAIL_GENETICS_HAIL_IMAGE', f'hailgenetics/hail:{pip_version()}')
+
+
+def init_parser(parser):
+    parser.add_argument('script', type=str, help='Path to script')
+    parser.add_argument('--name', type=str, default='', help='Batch name')
+    parser.add_argument('--image-name', type=str, required=False,
+                        help='Name for Docker image. Defaults to hailgenetics/hail')
+    parser.add_argument('--files', nargs='+', action='append', default=[],
+                        help='Comma-separated list of files or directories to add to the working directory of job')
+    parser.add_argument('-o', type=str, default='text', choices=['text', 'json'])
+
+
+async def async_main(args):
+    script = args.script
+    files = unpack_comma_delimited_inputs(args.files)
+    user_config = get_user_config_path()
+    quiet = args.o != 'text'
+
+    remote_tmpdir = get_remote_tmpdir('hailctl batch submit')
+
+    def cloud_prefix(path):
+        return f'{remote_tmpdir}/{path}'
```

Comment:
You must assume other programs are sharing `remote_tmpdir` with you. `get_remote_tmpdir` is just about shimming from the old configuration variables to a remote temporary directory. Just `sceret_alnum_string` from `hailtop.utils` to generate a random path component that you can use for your own purposes.